Pass the last day of the week as end date when fetching revenue stats api

anitaa1990 opened the pull request:
Fixes #1463 - This PR adds the option to pass the end of the week date as the end date, when fetching revenue stats from the `/wc/v4/reports/revenue/stats` endpoint.

#### To Test
- Click on `Woo` -> `Revenue Stats` -> `Select Site` -> `Fetch Current week revenue stats`.
- Verify that the end date for the weekly stats is the future date for the end of the week instead of the current date. i.e. for this week end date should be `28-12-2019`.


<img width="300" src="https://user-images.githubusercontent.com/22608780/71457445-00e06280-27c4-11ea-9953-c06d95d81633.png">

AmandaRiu commented on plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCStatsStore.kt:
```diff
@@ -674,6 +674,27 @@ class WCStatsStore @Inject constructor(
+    private fun getEndDateForRevenueStatsGranularity(
+        site: SiteModel,
+        granularity: StatsGranularity,
+        endDate: String?
```
What is the argument `endDate` for? Doesn't appear to be used.

anitaa1990 replied:
The idea was that if we allow the option to pass `endDate` in the UI side in the future, we can use that `endDate`, rather than having to make changes in FluxC side again. But it's not really used at the moment as you said and I foresee some changes we would need to make anyway to accommodate those changes so removed the redundant `endDate` from the method now in a530d04.

AmandaRiu commented on plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCStatsStore.kt:
```diff
@@ -674,6 +674,27 @@ class WCStatsStore @Inject constructor(
     }
 
+    /**
+     * Given a [endDate], formats the date based on the site's timezone in format yyyy-MM-dd'T'hh:mm:ss
```
Seems like two different jobs were intended for this method.  Comment probably needs to be updated.

anitaa1990 replied:
Since I removed the `endDate`, the comment was updated as well in a530d04

AmandaRiu replied:
@anitaa1990 Definitely better, but the comments make it sound like all that's happening is formatting a date, but the method is doing something different. Would it be more accurate to say this method does the following?
```
Returns the appropriate end date for the [site] and [granularity] provided to use for fetching revenue stats.
```

anitaa1990 replied:
👍 Fixed in 2cbf507

AmandaRiu commented on plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/utils/DateUtils.kt:
```diff
@@ -314,6 +314,13 @@ object DateUtils {
         return formatDate(DATE_TIME_FORMAT_START, cal.time)
     }
 
+    fun getLastDayOfCurrentWeekBySite(site: SiteModel): String {
```
np: I expect this method to return a mapping of last day by site due to the naming. Recommend changing to `getLastDayOfCurrentWeekForSite()`.

anitaa1990 replied:
👍 Done in b7700e3

anitaa1990 commented:
> found that the end date being sent has a time of 00:00:00 - shouldn't that be 23:59:59 so the results include the full last day of the week?

Good catch! Fixed in b7700e3

Fixed all code comments and ready for another round 👍

AmandaRiu commented:
@anitaa1990 Looks like there are some failing tests.

anitaa1990 commented:
@AmandaRiu this looks like it was due to a flaky test. I re-ran CircleCI and it's working now. PR ready for another round 👍
